src/Uno.Core/Collections/ImmutableList.cs

```diff
@@ -24,33 +26,203 @@ namespace Uno.Collections
 	/// <summary>
 	/// An immutable list implementation, designed for safe concurrent access.
 	/// </summary>
-	public class ImmutableList<T>
+	public class ImmutableList<T> : IImmutableList<T>
 	{
-		private T[] data;
+		private readonly T[] _data;
 		private static ImmutableList<T> _empty = new ImmutableList<T>();
 
 		/// <summary>
 		/// Creates an empty list
 		/// </summary>
 		public ImmutableList()
 		{
-			data = new T[0];
+			_data = new T[0];
 		}
 
 		/// <summary>
 		/// Provides an empty list
 		/// </summary>
-		public static ImmutableList<T> Empty {
-			get { return _empty; }
-		}
+		public static ImmutableList<T> Empty => _empty;
 
 		/// <summary>
 		/// Initializes the list with the provided array.
 		/// </summary>
 		/// <param name="data"></param>
 		public ImmutableList(T[] data)
 		{
-			this.data = data;
+			_data = new T[data.Length];
+			Array.Copy(data, _data, data.Length);
+		}
+
+		public IImmutableList<T> Clear()
+		{
+			return Empty;
+		}
+
+		public int IndexOf(T item, int index, int count, IEqualityComparer<T> equalityComparer)
+		{
+			var comparer = equalityComparer ?? EqualityComparer<T>.Default;
+			for (var i = index; i < index + count; i++)
```

Comment:
Possible OutOfRange. How does behave the ImmutableList from ImmutableCollections ?

Reply:
Just checked: yes, same behavior.